File: PST/FUEL Source Code/FUEL/Backup/FS/FileProcessing.cs
// Decompiled with JetBrains decompiler
// Type: FUEL.FS.FileProcessing
// Assembly: FUEL, Version=2.4.8.0, Culture=neutral, PublicKeyToken=null
// MVID: A3D630D0-0620-425D-8CFB-B67B2AC75507
// Assembly location: C:\hp\FUEL\FUEL.dll

using FUEL.My;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using Microsoft.VisualBasic.FileIO;
using System;
using System.Diagnostics;
using System.Text;
using System.Windows.Forms;

namespace FUEL.FS
{
  public class FileProcessing
  {
    [DebuggerNonUserCode]
    public FileProcessing()
    {
    }

    public static object ReadFile(string Path) => (object) MyProject.Computer.FileSystem.ReadAllText(FileSystem.SetPath(Path, FilePurpose.ForReading));

    public static string[,] ReadDelimitedFile(string Path, string Delimiter) => FileProcessing.ReadDelimitedFile(Path, Delimiter, (string[]) null, -1, -1L);

    public static string[,] ReadDelimitedFile(
      string Path,
      string Delimiter,
      string[] CommentTokens)
    {
      return FileProcessing.ReadDelimitedFile(Path, Delimiter, CommentTokens, -1, -1L);
    }

    private static string[,] ReadDelimitedFile(
      string Path,
      string Delimiter,
      string[] CommentTokens,
      int MaxFieldCnt,
      long LineCnt)
    {
      string path;
      try
      {
        path = FileSystem.SetPath(Path, FilePurpose.ForReading);
      }
      catch (Exception ex)
      {
        ProjectData.SetProjectError(ex);
        throw;
      }
      bool flag = false;
      string[,] strArray1 = (string[,]) null;
      if (MaxFieldCnt != -1)
      {
        flag = true;
        strArray1 = new string[checked ((int) (LineCnt - 1L) + 1), checked (MaxFieldCnt - 1 + 1)];
      }
      using (TextFieldParser textFieldParser = new TextFieldParser(path))
      {
        textFieldParser.SetDelimiters(Delimiter);
        textFieldParser.CommentTokens = CommentTokens;
        textFieldParser.TrimWhiteSpace = true;
        int index1 = 0;
        while (!textFieldParser.EndOfData)
        {
          string[] strArray2 = textFieldParser.ReadFields();
          if (flag)
          {
            int num = checked (strArray2.Length - 1);
            int index2 = 0;
            while (index2 <= num)
            {
              strArray1[index1, index2] = strArray2[index2];
              checked { ++index2; }
            }
            checked { ++index1; }
          }
          else if (strArray2.Length > MaxFieldCnt)
            MaxFieldCnt = strArray2.Length;
          if (textFieldParser.LineNumber != -1L)
            LineCnt = textFieldParser.LineNumber;
          else if (textFieldParser.LineNumber != -1L & LineCnt == -1L)
            LineCnt = 1L;
        }
      }
      return flag ? strArray1 : FileProcessing.ReadDelimitedFile(Path, Delimiter, CommentTokens, MaxFieldCnt, LineCnt);
    }

    public static void WriteDelimitedFile(
      string Path,
      string[] Header,
      string[,] Body,
      string Delimiter)
    {
      FileProcessing.WriteDelimitedFile(Path, Header, Body, Delimiter, true);
    }

    public static void WriteDelimitedFile(
      string Path,
      string[] Header,
      string[,] Body,
      string Delimiter,
      bool AppendWriteExistingFile)
    {
      string file = FileSystem.SetPath(Path, FilePurpose.ForWriting);
      StringBuilder stringBuilder = new StringBuilder();
      if (!MyProject.Computer.FileSystem.FileExists(file) & Header != null)
      {
        string str = Strings.Join(Header, Delimiter);
        stringBuilder.Append(str).AppendLine();
      }
      int num1 = Information.UBound((Array) Body);
      int index1 = 0;
      while (index1 <= num1)
      {
        string str = (string) null;
        int num2 = Information.UBound((Array) Body, 2);
        int index2 = 0;
        while (index2 <= num2)
        {
          str += Body[index1, index2];
          if (index2 < Information.UBound((Array) Body, 2))
            str += Delimiter;
          checked { ++index2; }
        }
        stringBuilder.Append(str).AppendLine();
        checked { ++index1; }
      }
      string text = stringBuilder.ToString();
      MyProject.Computer.FileSystem.WriteAllText(file, text, AppendWriteExistingFile);
    }

    public static void WriteToFile(string Filename, string TextToWrite, bool Append)
    {
      Filename = FileSystem.SetPath(Filename, FilePurpose.ForWriting);
      MyProject.Computer.FileSystem.WriteAllText(Filename, TextToWrite, Append);
    }

    public static void ZipFiles(string SourceDir, string OutputName) => FileProcessing.ZipFiles(SourceDir, OutputName, 0.0, false);

    public static void ZipFiles(string SourceDir, string OutputName, double MaxFileSize) => FileProcessing.ZipFiles(SourceDir, OutputName, MaxFileSize, false);

    public static void ZipFiles(
      string SourceDir,
      string OutputName,
      double MaxFileSize,
      bool DeleteFiles)
    {
      frmZipProgress frmZipProgress = new frmZipProgress(SourceDir, OutputName, DeleteFiles);
      frmZipProgress.MaxFileSize = checked ((long) System.Math.Round(MaxFileSize));
      IntPtr owner = PST.getOwner();
      if (owner != (IntPtr) -1)
      {
        int num1 = (int) frmZipProgress.ShowDialog((IWin32Window) new WindowWrapper(owner));
      }
      else
      {
        int num2 = (int) frmZipProgress.ShowDialog();
      }
    }
  }
}

File: PST/FUEL/src/FUEL/FS/FileProcessing.vb
Imports FUEL
Imports FUEL.My
Imports Microsoft.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports Microsoft.VisualBasic.FileIO
Imports System
Imports System.Text

Namespace FUEL.FS
    Public Class FileProcessing
        ' Methods
        Public Shared Function ReadDelimitedFile(ByVal Path As String, ByVal Delimiter As String) As String(0 To .,0 To .)
            Return FileProcessing.ReadDelimitedFile(Path, Delimiter, Nothing, -1, -1)
        End Function

        Public Shared Function ReadDelimitedFile(ByVal Path As String, ByVal Delimiter As String, ByVal CommentTokens As String()) As String(0 To .,0 To .)
            Return FileProcessing.ReadDelimitedFile(Path, Delimiter, CommentTokens, -1, -1)
        End Function

        Private Shared Function ReadDelimitedFile(ByVal Path As String, ByVal Delimiter As String, ByVal CommentTokens As String(), ByVal MaxFieldCnt As Integer, ByVal LineCnt As Long) As String(0 To .,0 To .)
            Dim path As String = Nothing
            Try 
                path = FileSystem.SetPath(Path, FilePurpose.ForReading)
            Catch exception1 As Exception
                Dim ex As Exception = exception1
                ProjectData.SetProjectError(ex)
                Dim exception As Exception = ex
                Throw
            End Try
            Dim flag As Boolean = False
            Dim strArray3(,) As String(0 To .,0 To .) = Nothing
            If (MaxFieldCnt <> -1) Then
                flag = True
                strArray3 = New String((CInt((LineCnt - 1)) + 1)  - 1, ((MaxFieldCnt - 1) + 1)  - 1) {}
            End If
            Dim objA As New TextFieldParser(path)
            Try 
                Dim delimiters As String() = New String() { Delimiter }
                objA.SetDelimiters(delimiters)
                objA.CommentTokens = CommentTokens
                objA.TrimWhiteSpace = True
                Dim num As Integer = 0
                Do While True
                    If objA.EndOfData Then
                        Exit Do
                    End If
                    Dim strArray As String() = objA.ReadFields
                    If Not flag Then
                        If (strArray.Length > MaxFieldCnt) Then
                            MaxFieldCnt = strArray.Length
                        End If
                    Else
                        Dim num3 As Integer = (strArray.Length - 1)
                        Dim index As Integer = 0
                        Do While True
                            Dim num4 As Integer = num3
                            If (index > num4) Then
                                num += 1
                                Exit Do
                            End If
                            strArray3(num, index) = strArray(index)
                            index += 1
                        Loop
                    End If
                    If (objA.LineNumber <> -1) Then
                        LineCnt = objA.LineNumber
                    ElseIf ((objA.LineNumber <> -1) And (LineCnt = -1)) Then
                        LineCnt = 1
                    End If
                Loop
            Finally
                If Not Object.ReferenceEquals(objA, Nothing) Then
                    DirectCast(objA, IDisposable).Dispose
                End If
            End Try
            Return If(Not flag, FileProcessing.ReadDelimitedFile(Path, Delimiter, CommentTokens, MaxFieldCnt, LineCnt), strArray3)
        End Function

        Public Shared Function ReadFile(ByVal Path As String) As Object
            Dim file As String = FileSystem.SetPath(Path, FilePurpose.ForReading)
            Return MyProject.Computer.FileSystem.ReadAllText(file)
        End Function

        Public Shared Sub WriteDelimitedFile(ByVal Path As String, ByVal Header As String(), ByVal Body As String(0 To .,0 To .), ByVal Delimiter As String)
            FileProcessing.WriteDelimitedFile(Path, Header, Body, Delimiter, True)
        End Sub

        Public Shared Sub WriteDelimitedFile(ByVal Path As String, ByVal Header As String(), ByVal Body As String(0 To .,0 To .), ByVal Delimiter As String, ByVal AppendWriteExistingFile As Boolean)
            Dim file As String = FileSystem.SetPath(Path, FilePurpose.ForWriting)
            Dim builder As New StringBuilder
            If (Not MyProject.Computer.FileSystem.FileExists(file) And (Not Header Is Nothing)) Then
                builder.Append(Strings.Join(Header, Delimiter)).AppendLine
            End If
            Dim num3 As Integer = Information.UBound(Body, 1)
            Dim num As Integer = 0
            Do While (num <= num3)
                Dim str4 As String = Nothing
                Dim num4 As Integer = Information.UBound(Body, 2)
                Dim num2 As Integer = 0
                Do While True
                    Dim num5 As Integer = num4
                    If (num2 > num5) Then
                        builder.Append(str4).AppendLine
                        num += 1
                        Exit Do
                    End If
                    str4 = (str4 & Body(num, num2))
                    If (num2 < Information.UBound(Body, 2)) Then
                        str4 = (str4 & Delimiter)
                    End If
                    num2 += 1
                Loop
            Loop
            MyProject.Computer.FileSystem.WriteAllText(file, builder.ToString, AppendWriteExistingFile)
        End Sub

        Public Shared Sub WriteToFile(ByVal Filename As String, ByVal TextToWrite As String, ByVal Append As Boolean)
            Filename = FileSystem.SetPath(Filename, FilePurpose.ForWriting)
            MyProject.Computer.FileSystem.WriteAllText(Filename, TextToWrite, Append)
        End Sub

        Public Shared Sub ZipFiles(ByVal SourceDir As String, ByVal OutputName As String)
            FileProcessing.ZipFiles(SourceDir, OutputName, 0, False)
        End Sub

        Public Shared Sub ZipFiles(ByVal SourceDir As String, ByVal OutputName As String, ByVal MaxFileSize As Double)
            FileProcessing.ZipFiles(SourceDir, OutputName, MaxFileSize, False)
        End Sub

        Public Shared Sub ZipFiles(ByVal SourceDir As String, ByVal OutputName As String, ByVal MaxFileSize As Double, ByVal DeleteFiles As Boolean)
            Dim progress As New frmZipProgress(SourceDir, OutputName, DeleteFiles) With { _
                .MaxFileSize = CLng(Math.Round(MaxFileSize)) _
            }
            Dim aHandle As IntPtr = PST.getOwner
            If (aHandle <> DirectCast(-1, IntPtr)) Then
                progress.ShowDialog(New WindowWrapper(aHandle))
            Else
                progress.ShowDialog
            End If
        End Sub

    End Class
End Namespace

